Sources/NostrSDK/Events/MuteListEvent.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  MuteListEvent.swift
+//
+//
+//  Created by Bryan Montz on 12/15/23.
+//
+
+import Foundation
+
+/// An event that contains various things the user doesn't want to see in their feeds.
+///
+/// See [NIP-51](https://github.com/nostr-protocol/nips/blob/master/51.md#standard-lists).
+public final class MuteListEvent: NostrEvent, HashtagInterpreting, DirectMessageEncrypting {
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+    
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+    
+    init(content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: .muteList, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+    
+    /// The publicly muted public keys (authors).
+    public var pubkeys: [String] {
```

Comment:
Wondering if it makes sense to have something like a `PubkeyTagInterpreting` as there's a few Nostr events that do it.

However, one issue that I see with refactoring in this way is that we lose some of the finer-grained documentation on the kind level.

Reply:
Agree. It would be funny if you could override *just* the documentation. I think it's a good idea to move to a protocol for interpreting, but I'll do this in a next PR for nothing but tags improvements.